### PR TITLE
feat(tab): Add MDCTab component

### LINKED ISSUE #2376: Create new `mdc-tab` component
- [ ] HTML + CSS for a tab with text (`.mdc-tab__text-label`)
- [ ] HTML + CSS for a tab with an icon (`.mdc-tab__icon`)
- [ ] HTML + CSS for a tab with text and an icon
- [ ] `mdc-tab-text-label-color` Sass mixin
- [ ] `mdc-tab-icon-color` Sass mixin
- [ ] Demo of customizing active and inactive state (`.mdc-tab--active`)
- [ ] `MDCTabFoundation` with activate and deactivate methods
- [ ] `MDCTab` vanilla version, with proxy to activate  and deactivate methods
- [ ] Publish new `mdc-tab` Node module

Demo should show a `mdc-tab` activating and deactivating, plus customization

### PR DESCRIPTION
**Pull request by patrickrodee**
Fixes #2376

### CODE REVIEW

File: packages/material-components-web/index.js [hunk 1]
```diff
@@ -56,6 +57,7 @@ autoInit.register('MDCLinearProgress', linearProgress.MDCLinearProgress);
 autoInit.register('MDCNotchedOutline', notchedOutline.MDCNotchedOutline);
 autoInit.register('MDCRadio', radio.MDCRadio);
 autoInit.register('MDCSnackbar', snackbar.MDCSnackbar);
+autoInit.register('MDCTab_', tab.MDCTab);
```
Comment (patrickrodee) [inline packages/material-components-web/index.js]: A working but imperfect solution to avoid naming collision.

File: packages/mdc-tab/index.js [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+  set active(isActive) {
+    if (isActive) {
+      this.foundation_.activate();
+    } else {
+      this.foundation_.deactivate();
+    }
+  }
+
+  /** @return {!MDCRipple} */
+  get ripple() {
```
Comment (lynnjepsen) [inline packages/mdc-tab/index.js]: do we need a `get ripple` method on MDCTab?

File: packages/mdc-tab/mdc-tab.scss [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+}
+
+.mdc-tab__icon + .mdc-tab__text-label {
+  @include mdc-rtl-reflexive-box(padding, left, 8px);
+}
+
+/**
+ * The [de]activation animation affects color for text label and icon
+ */
+.mdc-tab--animating-activate .mdc-tab__text-label,
```
Comment (lynnjepsen) [inline packages/mdc-tab/mdc-tab.scss]: should we just have an `mdc-tab--animating` class?

File: packages/mdc-tab/package.json [hunk 4]
```diff
@@ -0,0 +1,24 @@
+{
+  "name": "@material/tab",
+  "description": "The Material Components for the web tab component",
+  "version": "0.33.0",
```
Comment (lynnjepsen) [inline packages/mdc-tab/package.json]: this should be version 0.0.0 I think...then Lerna up the version number later.

I forget where the docs are, but somewhere there are docs saying how to create a new package.

File: test/build/goldens/build-config-dev-env.golden.json [hunk 5]
```diff
@@ -63,6 +63,7 @@
       "slider": "/demos/slider.scss",
       "snackbar": "/demos/snackbar.scss",
       "switch": "/demos/switch.scss",
+      "tab": "/demos/tab.scss",
```
Comment (lynnjepsen) [inline test/build/goldens/build-config-dev-env.golden.json]: Ask Andy if we still need these golden files. I think the webpack refactor is over...so maybe we can remove these files

File: packages/mdc-tab/mdc-tab.scss (hunk 3 above)
Comment (patrickrodee) [reply packages/mdc-tab/mdc-tab.scss]: The separate classes are because activation has a 100ms delay.

File: packages/mdc-tab/index.js (hunk 2 above)
Comment (patrickrodee) [reply packages/mdc-tab/index.js]: Not sure. I was following other components and exposing a getter for MDCRipple seems to be common practice.

But is it necessary? Probably not.

File: packages/mdc-tab/package.json (hunk 4 above)
Comment (patrickrodee) [reply packages/mdc-tab/package.json]: Dope, I'll check that out.

File: test/build/goldens/build-config-dev-env.golden.json (hunk 5 above)
Comment (patrickrodee) [reply test/build/goldens/build-config-dev-env.golden.json]: Confirmed with @acdvorak these are necessary for now.

File: packages/mdc-tab/mdc-tab.scss (hunk 3 above)
Comment (patrickrodee) [reply packages/mdc-tab/mdc-tab.scss]: Offline discussion: keep em

**Pull request merged**